src/python/WMCore/WMRuntime/Watchdog.py

```diff
@@ -78,6 +81,37 @@ def setupMonitors(self, task, wmbsJob):
                 #This should be a config section
                 monitorArgs = getattr(task.data.watchdog, monitor)
                 args = monitorArgs.dictionary_()
+                # Scale resources according to the HTCondor runtime environment.
+                origCores = 1
+                try:
+                    origCores = int(task.data.application.multicore.numberOfCores)
+                except AttributeError:
+                    print("No value for numberOfCores.  Assuming 1")
+                resources = {'cores': origCores}
+                origMaxRSS = args.get('maxRSS')
+                if origMaxRSS:
+                    resources['memory'] = origMaxRSS
+                # Actually parses the HTCondor runtime
+                resizeResources(resources)
+                # HTCondor doesn't explicitly scale VSize; hence we scale it
+                # according to the scaling of maxRSS between the workflow and runtime.
+                if origMaxRSS and ('maxVSize' in args):
+                    args['maxVSize'] *= resources['memory'] / origMaxRSS
```

Comment:
Yep, we decided not to kill anymore on VSize. We used to have 20GB per core in production.
I believe some protection is still good to have, in case things go very very bad.